aviary/utils/propeller_map_conversion.py

```diff
@@ -0,0 +1,252 @@
+
+
+def _read_gasp_propeller(fp, cmts):
+    """Read a GASP propeller map file and parse its scalars and tabular data.
+    Data table is returned as a dictionary.
+    The table consists of both the independent variables and the dependent variable for
+    the corresponding field. The table is a "tidy format" 2D array where the first three
+    columns are the independent varaiables (Advance ratio, Mach number, and power coefficient)
+    and the final column is the dependent variable thrust coefficient.
+    """
+    with open(fp, "r") as f:
+        table_types = ["thrust_coefficient",]
+        scalars = _read_pm_header(f)
+        if scalars['iread'] == 1:
+            cmts.append('# CT = f(Helical Mach at 75% Radius, Adv ratio & CP)')
+        elif scalars['iread'] == 2:
+            cmts.append('Propfan format - CT = f(Mach, Adv Ratio & CP)')
+        else:
+            raise RuntimeError(f"IREAD = 1 or 2 expected, got {scalars['iread']}")
+
+        tables = {k: _read_pm_table(f, cmts) for k in table_types}
+
+    return scalars, tables, table_types
+
+
+def _read_pm_header(f):
+    """Read GASP propeller map header, returning the propeller scalars in a dict"""
+    # parameter 2 is IPRINT in GASP and is ignored
+    iread, _ = _parse(
+        f, [*_rep(2, (int, 5))]
+    )
+
+    return {
+        "iread": iread,
+    }
+
+
+def _read_pm_table(f, cmts):
+    """Read an entire table from a GASP propeller map file.
+    The table data is returned as a "tidy format" array with three columns for the
+    independent variables (advanced ratio (J), Mach number and power coefficient) 
+    and the final column for thrust coefficient.
+    """
+    tab_data = None
+
+    # table title
+    title = f.readline().strip()
+    cmts.append(f'# {title}')
+    # number of maps in the table
+    (nmaps,) = _parse(f, [(int, 5)])
+    # blank line
+    f.readline()
+
+    for i in range(nmaps):
+        map_data = _read_map(f)
+
+        # blank line following all but the last map in the table
+        if i < nmaps - 1:
+            f.readline()
+
+        if tab_data is None:
+            tab_data = map_data
+        else:
+            tab_data = np.r_[tab_data, map_data]
+
+    return tab_data
+
+
+def _rep(n, t):
+    """Shorthand for ``itertools.repeat`` with the multiplier first."""
+    return itertools.repeat(t, n)
+
+
+def _parse(f, fmt):
+    """Read a line from file ``f`` and parse it according to the given ``fmt``"""
+    return _strparse(f.readline(), fmt)
+
+
+def _strparse(s, fmt):
+    """Parse a string into fixed-width numeric fields.
+    ``fmt`` should be a list of tuples specifying (type, length) for each field in
+    string ``s``. Use None for the type to skip (i.e. not yield) that field.
+    """
+    p = 0
+    for typ, length in fmt:
+        sub = s[p: p + length]
+        if typ is not None:
+            yield typ(sub)
+        p += length
+
+
+def _read_map(f):
```

Comment:
same question here. Is the only difference the comments and format used?

Reply:
Where should we move it?

Reply:
Moved to `conversion_utils.py`